core/src/main/scala/org/apache/spark/sql/delta/commands/ShowTableColumnsCommand.scala

```diff
@@ -0,0 +1,150 @@
+ * @param schemaName (optional when tableName exists) the schema name (database name) of the table
+ */
+case class ShowTableColumnsCommand(
+    tableName: Option[TableIdentifier],
+    path: Option[String],
+    schemaName: Option[String])
+  extends LeafRunnableCommand with DeltaCommand {
+
+  override val output: Seq[Attribute] = ExpressionEncoder[TableColumns]().schema.toAttributes
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val (basePath, tableMetadata) =
+      getPathAndTableMetadata(sparkSession, path, tableName, schemaName)
+    val deltaLog = basePath match {
+      case Some(basePath) => DeltaLog.forTable(sparkSession, basePath)
+      // basePath would be None only if found a view schema
+      case None => return getColumnsFromSchema(tableMetadata.get.schema)
+    }
+
+    recordDeltaOperation(deltaLog, "delta.ddl.showColumns") {
+      deltaLog.snapshot.version match {
+        case -1 =>
+          try {
+            getColumnsFromSchema(tableMetadata.get.schema)
+          } catch {
+            case _: NoSuchElementException =>
+              // Throw FileNotFoundException when the path doesn't exist since there may be a typo
+              if (!basePath.get.getFileSystem(deltaLog.newDeltaHadoopConf()).exists(basePath.get)) {
+                throw DeltaErrors.fileNotFoundException(basePath.get.toString)
+              }
+              // `SHOW COLUMNS` command is not supported for non-Delta table described by file path
+              throw DeltaErrors.fileFormatNotSupportedInShowColumns(basePath.get.toString)
+          }
+        case _ => getColumnsFromSchema(deltaLog.snapshot.schema)
+      }
+    }
+  }
+
+  private def getColumnsFromSchema(schema: StructType): Seq[Row] =
+    schema.fieldNames.map { x => toRow(TableColumns(x)) }.toSeq
+
+  /**
+   * Resolve `path` and `tableIdentifier` to get the underlying storage path if it's not a view,
+   * and its `CatalogTable` if it's a table. The caller will make sure either `path` or
+   * `tableIdentifier` is set but not both.
+   *
+   * If `path` is set, return it and an empty `CatalogTable` since it's a physical path. If
+   * `tableIdentifier` is set, we will try to see if it's a Delta data source path (such as
+   * `delta.<a table path>`). If so, we will return the path and an empty `CatalogTable`. Otherwise,
+   * we will use `SessionCatalog` to resolve `tableIdentifier`. If this still failed, we will try to
+   * see if it's a view. If so, we will return the view description with an empty path.
+   */
+  protected def getPathAndTableMetadata(
+      spark: SparkSession,
+      path: Option[String],
+      tableIdentifier: Option[TableIdentifier],
+      schemaName: Option[String]
+  ): (Option[Path], Option[CatalogTable]) = {
+    val lookupTable = schemaName match {
+      case None => tableIdentifier
+      case Some(db) => Some(TableIdentifier(tableIdentifier.get.identifier, Some(db)))
```

Comment:
`lookupTable` can also be None. 
Then if `path` is None, like `SHOW COLUMNS IN`: it will raise `org.apache.spark.sql.catalyst.parser.ParseException: 
mismatched input '<EOF>'` error.
If `path` is not None, like `SHOW COLUMNS IN '/path/to/file'`: Then this function will only return `path`